src/spikeinterface/generation/session_displacement_generator.py

```diff
@@ -0,0 +1,510 @@
+import copy
+
+from spikeinterface.generation.drifting_generator import (
+    generate_probe,
+    fix_generate_templates_kwargs,
+    calculate_displacement_unit_factor,
+)
+from spikeinterface.core.generate import (
+    generate_unit_locations,
+    generate_sorting,
+    generate_templates,
+)
+import numpy as np
+from spikeinterface.generation.noise_tools import generate_noise
+from spikeinterface.core.generate import setup_inject_templates_recording, _ensure_firing_rates
+from spikeinterface.core import InjectTemplatesRecording
+
+
+def generate_session_displacement_recordings(
+    num_units=250,
+    recording_durations=(10, 10, 10),
+    recording_shifts=((0, 0), (0, 25), (0, 50)),
+    non_rigid_gradient=None,
+    recording_amplitude_scalings=None,
+    shift_units_outside_probe=False,
+    sampling_frequency=30000.0,
+    probe_name="Neuropixel-128",
+    generate_probe_kwargs=None,
+    generate_unit_locations_kwargs=dict(
+        margin_um=20.0,
+        minimum_z=5.0,
+        maximum_z=45.0,
+        minimum_distance=18.0,
+        max_iteration=100,
+        distance_strict=False,
+    ),
+    generate_templates_kwargs=dict(
+        ms_before=1.5,
+        ms_after=3.0,
+        mode="ellipsoid",
+        unit_params=dict(
+            alpha=(150.0, 500.0),
+            spatial_decay=(10, 45),
+        ),
+    ),
+    generate_sorting_kwargs=dict(firing_rates=(2.0, 8.0), refractory_period_ms=4.0),
+    generate_noise_kwargs=dict(noise_levels=(12.0, 15.0), spatial_decay=25.0),
+    extra_outputs=False,
+    seed=None,
+):
+    """
+    Generate a set of recordings simulating probe drift across recording
+    sessions.
+
+    Rigid drift can be added in the (x, y) direction in `recording_shifts`.
+    These drifts can be made non-rigid (scaled dependent on the unit location)
+    with the `non_rigid_gradient` parameter. Amplitude of units can be scaled
+    (e.g. template signal removed by scaling with zero) by specifying scaling
+    factors in `recording_amplitude_scalings`.
+
+    Parameters
+    ----------
+
+    num_units : int
+        The number of units in the generated recordings.
+    recording_durations : list
+        An array of length (num_recordings,) specifying the
+        duration that each created recording should be.
```

Comment:
Worth saying the units?

```suggestion
        duration that each created recording should be, in seconds.
```